pkg/cmd/cli/cli_changefeed_create.go

```diff
@@ -0,0 +1,495 @@
+		}
+		o.startTs = oracle.ComposeTS(ts, logical)
+	}
+
+	return nil
+}
+
+// validate checks that the provided attach options are specified.
+func (o *createChangefeedOptions) validate(ctx context.Context) error {
+	if o.commonChangefeedOptions.sinkURI == "" {
+		return errors.New("Creating changefeed without a sink-uri")
+	}
+
+	if err := o.validateStartTs(ctx); err != nil {
+		return err
+	}
+
+	return o.validateTargetTs()
+}
+
+func (o *createChangefeedOptions) getInfo(ctx context.Context, cmd *cobra.Command) (*model.ChangeFeedInfo, error) {
+	_, captureInfos, err := o.etcdClient.GetCaptures(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	cdcClusterVer, err := version.GetTiCDCClusterVersion(captureInfos)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	cfg := config.GetDefaultReplicaConfig()
+
+	if !cdcClusterVer.ShouldEnableOldValueByDefault() {
+		cfg.EnableOldValue = false
+		log.Warn("The TiCDC cluster is built from an older version, disabling old value by default.",
+			zap.String("version", cdcClusterVer.String()))
+	}
+
+	if o.commonChangefeedOptions.sortEngine == model.SortUnified && !cdcClusterVer.ShouldEnableUnifiedSorterByDefault() {
+		o.commonChangefeedOptions.sortEngine = model.SortInMemory
+		log.Warn("The TiCDC cluster is built from an older version, disabling Unified Sorter by default",
+			zap.String("version", cdcClusterVer.String()))
+	}
+
+	if len(o.commonChangefeedOptions.configFile) > 0 {
+		if err := o.commonChangefeedOptions.strictDecodeConfig("TiCDC changefeed", cfg); err != nil {
+			return nil, err
+		}
+	}
+
+	if o.disableGCSafePointCheck {
+		cfg.CheckGCSafePoint = false
+	}
+
+	if o.commonChangefeedOptions.cyclicReplicaID != 0 || len(o.commonChangefeedOptions.cyclicFilterReplicaIDs) != 0 {
+		if !(o.commonChangefeedOptions.cyclicReplicaID != 0 && len(o.commonChangefeedOptions.cyclicFilterReplicaIDs) != 0) {
+			return nil, errors.New("invalid cyclic config, please make sure using " +
+				"nonzero replica ID and specify filter replica IDs")
+		}
+
+		filter := make([]uint64, 0, len(o.commonChangefeedOptions.cyclicFilterReplicaIDs))
+		for _, id := range o.commonChangefeedOptions.cyclicFilterReplicaIDs {
+			filter = append(filter, uint64(id))
+		}
+
+		cfg.Cyclic = &config.CyclicConfig{
+			Enable:          true,
+			ReplicaID:       o.commonChangefeedOptions.cyclicReplicaID,
+			FilterReplicaID: filter,
+			SyncDDL:         o.commonChangefeedOptions.cyclicSyncDDL,
+			// TODO(neil) enable ID bucket.
+		}
+	}
+
+	if !cfg.EnableOldValue {
+		sinkURIParsed, err := url.Parse(o.commonChangefeedOptions.sinkURI)
+		if err != nil {
+			return nil, cerror.WrapError(cerror.ErrSinkURIInvalid, err)
+		}
+
+		protocol := sinkURIParsed.Query().Get("protocol")
+		for _, fp := range forceEnableOldValueProtocols {
+			if protocol == fp {
+				log.Warn("Attempting to replicate without old value enabled. CDC will enable old value and continue.", zap.String("protocol", protocol))
+				cfg.EnableOldValue = true
```

Comment:
If we set `cfg.EnbaleOldValue` with `false` above in line: 200 because it is older version and set it with `true` here, what whould happend?